js_modules/dagster-ui/packages/ui-core/src/asset-selection/input/util.ts

```diff
@@ -66,3 +63,20 @@ export const getAttributesMap = (assets: AssetGraphQueryItem[]) => {
     code_location: codeLocations,
   };
 };
+
+const memoizedTag = weakMapMemoize((key: string, value: string) => ({
+  key,
+  value,
+}));
+
+export type Attribute = 'kind' | 'code_location' | 'group' | 'owner' | 'tag' | 'status' | 'key';
```

Comment:
If you wanted to, this could probably be `keyof ReturnType<typeof getAttributesMap>`. It might be nice to have an explicit type on `getAttributesMap` ensuring it returns the full set of attributes defined here